app/routes.py

```diff
@@ -1 +1,318 @@
+    slack_message = f"Somone just completed the task {task.title}"
+    # slack_client = WebClient(token=slack_token) #carries the token 
+
+    url = "https://slack.com/api/chat.postMessage"
+    slack_token = os.environ.get("SLACK_API_TOKEN")
+    data = {
+        "channel": "task-notifications",
+        "text": f"Someone just completed a task {task.title}",
+        "token": slack_token
+    }
+    response = requests.post(url, data = data)
+
+    return jsonify({
+        "task": task.task_to_dict()
+    }), 200
+
+#START OF GOAL
+
+#CREATE
+@goal_bp.route("", methods = ["POST"])
+def create_goal():
+    request_body = request.get_json()
+
+    if "title" not in request_body:
+        return jsonify({
+            "details": "Invalid data"
+        }
+        ), 400
+    
+    new_goal = Goal.from_dict(request_body)
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return jsonify(
+        {
+            "goal": new_goal.goal_to_dict()
+        }), 201
+
+
+#GET ALL
+@goal_bp.route("", methods = ["GET"])
+def get_all_goals():
+    response = []
+
+    goals = Goal.query.all()
+
+    for each_goal in goals:
+        response.append(each_goal.goal_to_dict())
+
+    return jsonify(response), 200
+
+#GET ONE
+@goal_bp.route("/<goal_id>", methods=["GET"])
+def get_one_goal(goal_id):
+
+    goal = validate_goal(Goal, goal_id)
+    
+
+    return jsonify({
+        "goal": goal.goal_to_dict()
+    }), 200
+
+#UPDATE
+@goal_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+
+    goal = validate_goal(Goal, goal_id)
+    request_data = request.get_json()
+
+    goal.title = request_data["title"]
+
+    db.session.commit()
+
+    return jsonify({
+        "goal": goal.goal_to_dict()
+    }), 200
+
+#DELETE
+@goal_bp.route("/<goal_id>", methods=["DELETE"])
+def delete_goal(goal_id):
+
+    goal = validate_goal(Goal, goal_id)
+
+    db.session.delete(goal)
+    db.session.commit()
+
+    return{
+        "details": 'Goal 1 "Build a habit of going outside daily" successfully deleted'
+    }, 200
+
+
+#ONE TO MANY
+@goal_bp.route("/<int:goal_id>/tasks", methods=["POST"])
+def get_tasks_list(goal_id):
+
+    goal = validate_goal(Goal, goal_id) #goal_id 7
+
+    request_data = request.get_json()
+    task_ids_list = request_data["task_ids"]  #[1,2,3]
+
+    for each_task_id in task_ids_list:
+        task_obj = Task.query.get(each_task_id) #get elem(task_obj) in database
+        
+     
+        task_obj.goal_id = goal_id #set the goal_id as the task_data's goal_id
+        db.session.commit()
+     
+    return jsonify(
+        {
+            "id": goal_id,
+            "task_ids": task_ids_list
+        }
+    ), 200
+
+#GET GOAL
+@goal_bp.route("/<goal_id>/tasks", methods = ["GET"])
+def get_tasks_for_specific_goal(goal_id):
+    goal = validate_goal(Goal, goal_id)
+
+#this returns a list of all tasks instances that has the goal_id
+    tasks_obj_list = Task.query.filter_by(goal_id = goal_id).all()
```

Comment:
This is not necessary. The Goal class has an attribute called `tasks`, which is a list of Task objects tied to the specific goal. We just need to iterate over that list.